README.md

```diff
@@ -3,13 +3,57 @@
+our program's fisrt run at 1000 elements warmed up the CPU to an extent where it is underclocing to keep itself safe, this only happend because-
+you never clean your computer + your running a 350W threadripper with a stock cooler resting naked held securely by gravity ontop of your cpu. ![my computer before my real cooler arived](image-1.png)
+Now that the scene has been set, your seccond script, this time running a larger test of 10,000 elements, loads into memory. But tragicly it overflows into
+the swapfile! (as we only put 1 stick of 2 gb ram in our system...) The 50mb floppy disk we use as our prefered swapfile location is just enough to 
+pick up the slack and 100 secconds later after HARD but severely throttled number crunching (mostly disk reads) we calculate our final answer... 42!
+
+3. If we always ignore constant factors and mostly care about leading terms, then they can add up and lead to a significant discrepancy in our runtime.
+
 - Suppose finding a particular element in a binary search tree with 1,000
   elements takes 5 seconds. Given what you know about the asymptotic complexity
   of search in a binary search tree, how long would you guess finding the same
   element in a search tree with 10,000 elements takes? Explain your reasoning.
 
+i would say that the worst case would take:
+![6.66](calculator.png)
+6.6666 secconds
+
+reasoning: i set it up like ratio and solved for the missing variable. I assumed that-
+the scaling holdes and that there are no other variables affecting the runtime, ANd ig that it was the worst case i was calculating.
+I am assuming the complexity of the function to be the same as a properly balanced search tree where we cut the search space in half at each node.
+
 - You measure the time with 10,000 elements and it takes 100 seconds! List 3
   reasons why this could be the case, given that reasoning with the asymptotic
   complexity suggests a different time.
 
+1. reason: binary search trees arnt alwasy balanced. What i was trying to say was that
+trees have a best case of log(n) and a worst case of n. The realworld runtime for a real 
+data set is likely inbetween these two bounds if the dataset is not structured properly. in this case sorted.
+
+in my last responce i tried to talk about how the small input of 1,000 was shifting though around 200 elements
+a second, while the 10,000 was only doing 100. This means to me that the small input was not properly balanced
+because it should have been 500 elements the first second, 250 the 2nd, 125, the third second and so on. The bigger 
+input however is the real issue, because i would have expected it to be 5,000 elements disregarded during the first 
+second, 2,500 in the second, and 1,250 the third... but we only saw an average of 100... thats really bad.
+So my point is that the 2nd input is a REALLY bad tree while the first one is only a pretty bad tree.
```

Comment:
Still not following you -- how can you infer the number of elements ignored from the runtime? How can you derive from a single runtime that the tree wasn't balanced?

Reply:
As far as I can see you wouldn't get this runtime even if you assume the absolute best for the small input and the absolute worst for the large input (i.e. maximal difference).